cmd/ctr/commands/run/run_unix.go

```diff
@@ -465,8 +465,8 @@ func getNetNSPath(_ gocontext.Context, task containerd.Task) (string, error) {
 // It also provides a way to override the CDI spec file paths if required.
 func withStaticCDIRegistry() oci.SpecOpts {
 	return func(ctx gocontext.Context, _ oci.Client, _ *containers.Container, s *oci.Spec) error {
-		registry := cdi.GetRegistry(cdi.WithAutoRefresh(false))
-		if err := registry.Refresh(); err != nil {
+		_ = cdi.Configure(cdi.WithAutoRefresh(false))
+		if err := cdi.GetDefaultCache().Refresh(); err != nil {
```

Comment:
nit: you could replace this with the simpler `if err := cdi.Refresh(); err != nil {`